src/utils.rs

```diff
@@ -268,6 +270,69 @@ macro_rules! map_enum {
     };
 }
 
+pub unsafe fn string_view_into_str<'a>(string_view: native::WGPUStringView) -> Option<&'a str> {
+    if string_view.data.is_null() {
+        match string_view.length {
+            crate::conv::WGPU_STRLEN => None,
+            0 => Some(""),
+            _ => panic!("Null address to WGPUStringView!"),
+        }
+    } else {
+        let bytes = match string_view.length {
+            crate::conv::WGPU_STRLEN => CStr::from_ptr(string_view.data).to_bytes(),
+            _ => make_slice(string_view.data as *const u8, string_view.length),
+        };
+
+        Some(std::str::from_utf8_unchecked(bytes))
```

Comment:
Important to note: this doesn't check whether strings passed into the library are valid UTF-8. I previously many locations did, but I figured this wasn't necessary.